pyo3-macros-backend/src/params.rs

```diff
@@ -50,11 +61,21 @@ impl Holders {
     pub fn check_gil_refs(&self) -> TokenStream {
         self.gil_refs_checkers
             .iter()
-            .map(|e| quote_spanned! { e.span() => #e.function_arg(); })
+            .map(|(e, checker_type)| match checker_type {
+                GilRefCheckerType::FunctionArg => quote_spanned! { e.span() => #e.function_arg(); },
+                GilRefCheckerType::FromPyWith => {
+                    quote_spanned! { e.span() => #e.from_py_with_arg(); }
+                }
+            })
             .collect()
     }
 }
 
+enum GilRefCheckerType {
+    FunctionArg,
```

Comment:
I understand it is not required to model the state space (yet), but maybe the code would become clearer if the `syn::Ident` is passed into the variants, e.g.

```rust
enum GilRefChecker {
  FunctionArg(syn::Ident),
  FromPyWith(syn::Ident),
}
```

Reply:
Yeah, I thought about it as well, when I wrote it, but wasn't sure which one is the better option. I'll update it.